saleor/dashboard/product/forms.py

```diff
@@ -2,77 +2,161 @@
+from django.utils.text import slugify
+from django.utils.translation import pgettext_lazy
 
-from ...product.models import (ProductImage, Product, ShirtVariant, BagVariant,
-                               Shirt, Bag)
+from django_prices.templatetags.prices_i18n import gross
+
+from ...product.models import (ProductImage, Stock, ProductVariant, Product,
+                               ProductAttribute, AttributeChoiceValue)
 
 PRODUCT_CLASSES = {
-    'shirt': Shirt,
-    'bag': Bag}
+    Product: 'Default'
+}
 
 
-class ProductClassForm(forms.Form):
-    cls_name = forms.ChoiceField(
-        choices=[(name, name.capitalize()) for name in PRODUCT_CLASSES.keys()])
+def get_verbose_name(model):
+    return model._meta.verbose_name
 
 
-class ProductForm(forms.ModelForm):
-    class Meta:
-        model = Product
-        fields = ['name', 'description', 'collection']
+class ProductClassForm(forms.Form):
+    product_cls = forms.ChoiceField(
+        label=pgettext_lazy('Product class form label', 'Product class'),
+        widget=forms.RadioSelect,
+        choices=[(cls.__name__, presentation) for cls, presentation in
+                 PRODUCT_CLASSES.iteritems()])
+
+    def __init__(self, *args, **kwargs):
+        super(ProductClassForm, self).__init__(*args, **kwargs)
+        self.fields['product_cls'].initial = PRODUCT_CLASSES.keys()[0].__name__
 
 
-class ShirtForm(ProductForm):
+class StockForm(forms.ModelForm):
     class Meta:
-        model = Shirt
+        model = Stock
         exclude = []
+        widgets = {
+            'product': forms.HiddenInput()
+        }
 
+    def __init__(self, *args, **kwargs):
+        product = kwargs.pop('product')
+        super(StockForm, self).__init__(*args, **kwargs)
+        variants = product.variants.all()
+        if variants:
+            self.fields['variant'].choices = [(variant.pk, variant) for variant
+                                              in variants]
+        else:
+            self.fields['variant'].widget.attrs['disabled'] = True
 
-class BagForm(ProductForm):
+
+class ProductForm(forms.ModelForm):
     class Meta:
-        model = Bag
+        model = Product
         exclude = []
 
+    def __init__(self, *args, **kwargs):
+        super(ProductForm, self).__init__(*args, **kwargs)
+        self.fields['name'].widget.attrs['placeholder'] = pgettext_lazy('Product form labels', 'Give your awesome product a name')
+        self.fields['categories'].widget.attrs['data-placeholder'] = pgettext_lazy('Product form labels', 'Search')
+        self.fields['attributes'].widget.attrs['data-placeholder'] = pgettext_lazy('Product form labels', 'Search')
+
 
-class ImageInputWidget(ClearableFileInput):
-    url_markup_template = '<a href="{0}"><img src="{0}" width=50 /></a>'
+class ProductVariantForm(forms.ModelForm):
+    class Meta:
+        model = ProductVariant
+        exclude = ['attributes']
+        widgets = {
+            'product': forms.HiddenInput(),
```

Comment:
Why is there a `product` field here? Can I move a variant to a different product?